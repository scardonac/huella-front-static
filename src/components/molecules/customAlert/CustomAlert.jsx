//Icons
import { Icons } from "../../../assets/icons/IconProvider";

const { IconAlertWhite } = Icons; //Iconos
export const CustomAlert = ({ message, type }) => {

    let containerStyle = {
        success: 'bg-green-600',
        error: 'bg-red-600',
        warning: 'bg-yellow-600',
        info: 'bg-blue-600',
    }

    return (
        <div className={`${containerStyle[type]} rounded h-[28px] flex items-center mt-1 whitespace-normal w-fit p-2`}>
            <img src={IconAlertWhite} alt="Icono de alerta" />
            <p className="ml-1 text-white text-left text-base leading-5 tracking-tighter font-normal opacity-100">
                {message}
            </p>
        </div>
    )
}