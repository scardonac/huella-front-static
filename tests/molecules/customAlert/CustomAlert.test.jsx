import React from 'react';
import { render } from '@testing-library/react';
import { CustomAlert } from '../../../src/components/molecules/customAlert/customAlert';
//mocks svg
jest.mock('../../../src/assets/icons/IconProvider', () => ({
    Icons: {
        IconAlertWhite: 'IconAlertWhite',
    }
}));
describe('Pruebas en <CustomAlert />', () => {

    test('Debe renderizarse correctamente', () => {
        const { baseElement } = render(<CustomAlert message="Test" type="success" />);
        expect(baseElement).toMatchSnapshot();
    });

});


