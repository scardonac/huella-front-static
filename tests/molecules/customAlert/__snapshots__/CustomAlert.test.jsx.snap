// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <CustomAlert /> Debe renderizarse correctamente 1`] = `
<body>
  <div>
    <div
      class="bg-green-600 rounded h-[28px] flex items-center mt-1 whitespace-normal w-fit p-2"
    >
      <img
        alt="Icono de alerta"
        src="IconAlertWhite"
      />
      <p
        class="ml-1 text-white text-left text-base leading-5 tracking-tighter font-normal opacity-100"
      >
        Test
      </p>
    </div>
  </div>
</body>
`;
